pkg/daemon/ceph/client/radosnamespace.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/util/exec"
+)
+
+// CreateRadosNamespace create a rados namespace in a pool.
+// poolName is the name of the ceph block pool, the same as the CephBlockPool CR name.
+func CreateRadosNamespace(context *clusterd.Context, clusterInfo *ClusterInfo, poolName, namespaceName string) error {
+	logger.Infof("creating rados namespace %q", namespaceName)
+	//  rbd namespace create pool-name/namespace-name
+	args := []string{"namespace", "create", "--pool", poolName, "--namespace", namespaceName}
+	cmd := NewRBDCommand(context, clusterInfo, args)
+	cmd.JsonOutput = false
+	output, err := cmd.Run()
+	if err != nil {
+		code, ok := exec.ExitStatus(err)
+		if ok && code == int(syscall.EEXIST) {
+			logger.Infof("rados namespace %s/%s already exists", poolName, namespaceName)
```

Comment:
The rook operator can operate on multiple resources in multiple namespaces. All info/debug messages must list the namespaced name so context is clear.

For which cephcluster?

Reply:
Kubernetes resources are usually given in namespace/name format. Any instance of a thing/thing in logs will likely imply to readers that this is a kubernetes resource. I would suggest reworking the message to be something like `rados namespace X already exists in pool Y` so it does not imply this is a k8s resource.

Reply:
This is the standard followed in ceph also to log the namespace and the pool. do we really need to change the logging in all places?